flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -504,8 +522,11 @@ private void checkResourceRequirementsWithDelay() {
                                             Preconditions.checkNotNull(requirementsCheckFuture)
                                                     .complete(null);
                                         }),
-                        requirementsCheckDelay.toMillis(),
+                        delay,
                         TimeUnit.MILLISECONDS);
+                isRequirementsCheckLongDelay = false;
+            } else {
+                checkResourceRequirements();
```

Comment:
I have tried this way, but I found it will change this method's behavior to async when delay is 0, now the behavior is sync. So I keep the current implementation and most test do not have to be touched.

Reply:
BTW, I found that the two slot manager have the exactly same method `checkResourceRequirementsWithDelay`. Do we need to add a abstract class to put it to avoid the duplicated methods ?

Reply:
> I have tried this way, but I found it will change this method's behavior to async when delay is 0, now the behavior is sync. So I keep the current implementation and most test do not have to be touched.

I don't think this method by design is guaranteed to be triggered synchronously when the delay is zero.

It's always good to do deduplication work. But I think the `DeclarativeSlotManager` will be entirely replaced by the `FineGrainedSlotManager` in the near future. So, we do not need to do it atm.

Reply:
I mean most test around `DeclarativeSlotManager` test it as synchronously. It will be easy to let it be test with the old behavior. But I can replace it with a slight timeout wait. I will push a commit for it.

Reply:
What do you mean by "replace it with a slight timeout wait"?

Reply:
I mean some test reply on the synchronously behavior of the DeclarativeSlotManager#checkResourceRequirements. So it will check the result in a synchronously way. "replace it with a slight timeout wait" means to replace these `Assert` with a slight timeout wait.

Reply:
I think it is a bit tricky and will harm the test stability.

Reply:
I'm lean to keep the current implementation and simply that logic after the retirement of the `DeclarativeSlotManager `.

Reply:
👌🏻